main.py

```
@@ -197,6 +197,14 @@ async def handler(data):
             return
 
         if msg_type == "hello":
+            if not isinstance(payload, dict):
+                logger.warning(
+                    "Malformed hello from %s: payload is not a dict (got %s). Disconnecting.",
+                    peer_addr, type(payload).__name__
+                )
+                asyncio.create_task(network.disconnect_peer(peer_addr))
```

Ruff (0.16.1)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)